src/Uno.UI/UI/Xaml/UIElement.Android.cs

```diff
@@ -22,29 +22,55 @@ namespace Windows.UI.Xaml
 	public partial class UIElement : BindableView
 	{
 		/// <summary>
-		/// Returns true if this element has children and they are all native (non-UIElements), false if it has no children or if at
-		/// least one is a UIElement.
+		/// Keeps the count of native children (non-UIElements), for clipping purpoess.
 		/// </summary>
-		private bool AreChildrenNativeViewsOnly
+		private int _nativeChildrenCount;
+		private bool _nativeClipChildren;
+
+		private Rect _previousClip = Rect.Empty;
+
+		private void ComputeAreChildrenNativeViewsOnly()
 		{
-			get
+			var nativeClipChildren = (this as IShadowChildrenProvider).ChildrenShadow.Count == _nativeChildrenCount;
+
+			if (_nativeClipChildren != nativeClipChildren)
 			{
-				var shadow = (this as IShadowChildrenProvider).ChildrenShadow;
-				if (shadow.Count == 0)
-				{
-					return false;
-				}
+				_nativeClipChildren = nativeClipChildren;
 
-				foreach (var child in shadow)
-				{
-					if (child is UIElement)
-					{
-						return false;
-					}
-				}
+				// Non-UIElements typically expect to be clipped, and display incorrectly otherwise
+				// This won't work when UIElements and non-UIElements are mixed in the same Panel,
```

Comment:
This comment doesn't make sense any more - IFrameworkElement is internal and it's now not possible to have non-UIElements as direct children of a Panel